atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -377,6 +378,14 @@ enum TransactionType {
     @Idempotent
     void abort();
 
+    /**
+     * Pre-commit hooks that are run on commit() just before the transaction is closed.
+     * Consumers can perform any actions as they wish - e.g. reading new data or writing to tables.
+     *
+     * @param preCommitAction the lambda executed just before commit
+     */
+    void preCommit(Runnable preCommitAction);
```

Comment:
I'm assuming that this as a hook is useful for other things you had in mind?

Reply:
ya, in general is a useful hook

Reply:
Let's put this in a separate PR if you want to ship this.